File: Dagger2TestAndroid/app/src/main/java/com/example/raghu/dagger2testandroid/dagger/ActivityBuilder.java
package com.example.raghu.dagger2testandroid.dagger;

import com.example.raghu.dagger2testandroid.ui.main.MainActivity;
import com.example.raghu.dagger2testandroid.ui.main.MainActivityModule;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

/**
 * Created by raghu on 4/8/17.
 */

@Module
public abstract class ActivityBuilder {

    @ActivityScope
    @ContributesAndroidInjector(modules = MainActivityModule.class)
    abstract MainActivity bindMainActivity();


}

File: SpecBeeMapAssignment/app/src/main/java/com/example/raghu/specbeemapassignment/dagger/ActivityBuilder.kt
package com.example.raghu.dagger2testandroid.dagger



import com.example.raghu.specbeemapassignment.MainActivity
import com.example.raghu.specbeemapassignment.dagger.MainActivityModule
import dagger.Module
import dagger.android.ContributesAndroidInjector

/**
 * Created by raghu on 4/8/17.
 */

@Module
abstract class ActivityBuilder {

    @ContributesAndroidInjector(modules = arrayOf(MainActivityModule::class))
    internal abstract fun bindMainActivity(): MainActivity


}